src/backend/aspen/api/views/samples.py

```diff
@@ -1,27 +1,188 @@
+            sample_gisaid_tuple_map[sample.id] = SampleGisaidTuple(
+                sample, sample.uploaded_pathogen_genome, None, None
+            )
+            genome_id_to_sample_id[
+                sample.uploaded_pathogen_genome.entity_id
+            ] = sample.id
+        else:
+            sample_gisaid_tuple_map[sample.id] = SampleGisaidTuple(
+                sample, None, None, None
+            )
+
+    gisaid_accession_workflows_inputs_query = (
+        sa.select(GisaidAccessionWorkflow)  # type: ignore
+        .join(GisaidAccessionWorkflow.inputs)  # type: ignore
+        .filter(Entity.id.in_(genome_id_to_sample_id.keys()))
+        .options(
+            selectinload(GisaidAccessionWorkflow.inputs),
+        )
+    )
+    gisaid_accession_workflows_with_inputs_response = await db.execute(
+        gisaid_accession_workflows_inputs_query
+    )
+    gisaid_accession_workflows_with_inputs: List[
+        GisaidAccessionWorkflow
+    ] = gisaid_accession_workflows_with_inputs_response.scalars().all()
+
+    # get around circular references
+    gisaid_accessions_query = sa.select(GisaidAccession).filter(  # type: ignore
+        GisaidAccession.producing_workflow_id.in_(  # type: ignore
+            [
+                workflow.workflow_id
+                for workflow in gisaid_accession_workflows_with_inputs
+            ]
+        )
+    )
+    gisaid_accessions_result = await db.execute(gisaid_accessions_query)
+    gisaid_accessions = gisaid_accessions_result.scalars().all()
+
+    workflow_to_accession_map: Dict[int, GisaidAccession] = dict()
+    for accession in gisaid_accessions:
+        workflow_to_accession_map[accession.producing_workflow_id] = accession
+
+    for gisaid_accession_workflow in gisaid_accession_workflows_with_inputs:
+        # A GisaidAccessionWorkflow is only ever going to have one input,
+        # a single UploadedPathogenGenome. The only way to create the former
+        # is through a method on the latter, and no other inputs are passed.
+        # The relationship is effectively one-to-one, despite the current db
+        # schema that technically allows for a many-to-many relationship.
+        # This is the same for its relationship outputs.
+        genome_id = gisaid_accession_workflow.inputs[0].entity_id
+        sample_id = genome_id_to_sample_id[genome_id]
+        origin_tuple = sample_gisaid_tuple_map[sample_id]
+        gisaid_accession = workflow_to_accession_map.get(
+            gisaid_accession_workflow.workflow_id, None
+        )
+        sample_gisaid_tuple_map[sample_id] = SampleGisaidTuple(
+            origin_tuple.sample,
+            origin_tuple.uploaded_pathogen_genome,
+            gisaid_accession_workflow,
+            gisaid_accession,
+        )
+
+    # populate sample object according to response schema
+    results: MutableSequence[Mapping[str, Any]] = list()
+    for sample_gisaid_tuple in sample_gisaid_tuple_map.values():
```

Comment:
I think I learned how do do (most) of this with pydantic models - want to see if we can do that here too?